src/AspNetCoreModuleV2/InProcessRequestHandler/inprocessapplication.cpp

```diff
@@ -220,6 +220,10 @@ IN_PROCESS_APPLICATION::ExecuteApplication()
         // set the callbacks
         s_Application = this;
 
+        // Set the current working directory to the application path.
+        // This keeps the current directory behavior consistent with out of process
+        SetCurrentDirectory(QueryApplicationPhysicalPath().c_str());
```

Comment:
Should current directory be reset after CLR is started, to preserve compat for other IIS modules?

Reply:
I'm not sure if we should even take this change. We can't reset current directory after CLR started because we don't know which user code relies on it and it would even more confusing if it changes during runtime.

Reply:
Also, the core clr calls into native to query the current directory. We can't set it after startup. It's the same issue that console redirection had.